packages/core/src/zip/__tests__/select.js

```diff
@@ -0,0 +1,238 @@
+'use strict'
+
+import I from 'immutable'
+import R from 'ramda'
+
+import { elementZipper } from '../'
+import { childrenProperty, isElement, isOfKind } from '../../data'
+
+import {
+  isStringArray,
+  isLocationArray,
+  child,
+  ancestors,
+  descendants,
+  children,
+  propEq,
+  select,
+  ofKind,
+} from '../select'
+
+const zipper = data => elementZipper({})(data)
+
+const data = {
+  kind: ['navigation', 'tabs'],
+  [childrenProperty]: ['tabs', 'settings'],
+  title: 'App',
+  tabs: [
+    {
+      kind: ['tab'],
+      title: 'Murder Bot',
+    },
+    {
+      kind: ['tab'],
+      title: 'Dr Mensah',
+      [childrenProperty]: ['elements', 'deeplink'],
+      elements: [
+        {
+          kind: ['element'],
+          title: 'robot',
+        },
+        {
+          kind: ['element'],
+          title: 'human',
+        },
+      ],
+      deeplink: {
+        kind: ['link'],
+        title: 'goodreads',
+      },
+    },
+  ],
+  settings: {
+    kind: ['settings'],
+    title: 'Martha Wells',
+  },
+}
+
+describe('zip.select', () => {
+  describe('helpers', () => {
+    it('isStringArray', () => {
+      expect(isStringArray(['string', 'test'])).toEqual(true)
+      expect(isStringArray(['string', null])).toEqual(false)
+      expect(isStringArray(['string', undefined])).toEqual(false)
+      expect(isStringArray([{}, 'test'])).toEqual(false)
+      expect(isStringArray(['string', []])).toEqual(false)
+      expect(isStringArray([[]])).toEqual(false)
+      expect(isStringArray([])).toEqual(true)
+      expect(isStringArray(null)).toEqual(false)
+      expect(isStringArray(undefined)).toEqual(false)
+    })
+    it('isLocationArray', () => {
+      expect(isLocationArray()).toEqual(false)
+      expect(isLocationArray(null)).toEqual(false)
+      expect(isLocationArray(undefined)).toEqual(false)
+      expect(isLocationArray([])).toEqual(false)
+      expect(isLocationArray(['test'])).toEqual(false)
+      expect(isLocationArray([zipper(I.fromJS(data))])).toEqual(true)
+      expect(
+        isLocationArray([zipper(I.fromJS(data)), zipper(I.fromJS(data))])
+      ).toEqual(true)
+      expect(
+        isLocationArray([
+          zipper(I.fromJS(data)),
+          'Not a Zipper',
+          zipper(I.fromJS(data)),
+        ])
+      ).toEqual(false)
+    })
+  })
+
+  describe('predicates', () => {
+    it('child', () => {
+      const root = zipper(I.fromJS(data))
+
+      const storage = child('storage')(root)
+      expect(storage).toBeNull()
+
+      // works for child pointing to an element
+      const settings = child('settings')(root).value()
+      expect(isElement(settings)).toEqual(true)
+      expect(settings.get('title')).toEqual('Martha Wells')
+
+      // doesn't work for child pointing to a collection
+      // TODO Andon: Check if this is correct (desired) behavior
```

Comment:
You could re-redefine `child` to a motion that moves you to the first child under a `childCollection`. This would make the function well defined (won't return null unless there are no children to be found at the specified collection),  and for most purposes it will work as desired.